game/src/ecs/systems/HealthSystem.java

```diff
@@ -0,0 +1,64 @@
+package ecs.systems;
+
+import ecs.components.AnimationComponent;
+import ecs.components.HealthComponent;
+import ecs.components.MissingComponentException;
+import ecs.damage.Damage;
+import ecs.entities.Entity;
+import java.util.List;
+import mydungeon.ECS;
+
+/**
+ * The HealthSystem offsets the damage to be done to all entities with the HealthComponent. Triggers
+ * the death of an entity when the hitpoints have fallen below 0.
+ */
+public class HealthSystem extends ECS_System {
+    @Override
+    public void update() {
+        for (Entity entity : ECS.entities) {
+            entity.getComponent(HealthComponent.name)
+                    .ifPresent(
+                            component -> {
+                                {
+                                    final AnimationComponent ac =
+                                            (AnimationComponent)
+                                                    entity.getComponent(AnimationComponent.name)
+                                                            .orElseThrow(
+                                                                    () ->
+                                                                            new MissingComponentException(
+                                                                                    AnimationComponent
+                                                                                            .name));
+
+                                    HealthComponent hpComponent = (HealthComponent) component;
+
+                                    // is the entity dead?
+                                    if (hpComponent.getCurrentHitPoints() <= 0) {
```

Comment:
Refactoring: Extract Method. Bitte auslagern - dann kannst Du hier ganz einfach eine Methodenreferenz in den Stream werfen. Und das Testen wird leichter.

Reply:
Ich hab jetzt was gemacht (das fängt ja schon gut an), glaube aber dich falsch verstanden zu haben.